Services.Test/DeviceModelsGenerationTest.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Moq;
+using System.Collections.Generic;
+using DeviceModel = Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models.DeviceModel;
+using Xunit.Abstractions;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services;
+using static Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models.Simulation;
+using Xunit;
+using Services.Test.helpers;
+using System;
+using static Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models.DeviceModel;
+
+namespace Services.Test
+{
+    public class DeviceModelsGenerationTest
+    {
+        /// <summary>The test logger</summary>
+        private readonly ITestOutputHelper log;
+
+        private Mock<ILogger> logger;
+        private DeviceModelsGeneration target;
+
+        public DeviceModelsGenerationTest(ITestOutputHelper log)
+        {
+            this.logger = new Mock<ILogger>();
+            this.target = new DeviceModelsGeneration(this.logger.Object);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void OverrideWithoutMessageSchema()
+        {
+            // Arrange
+            var source = new DeviceModel
+            {
+                Telemetry = new List<DeviceModelMessage>
+                {
+                    new DeviceModelMessage()
+                    {
+                        Interval = TimeSpan.Zero,
+                        MessageTemplate = string.Empty,
+                        MessageSchema = new DeviceModelMessageSchema()
+                        {
+                            Name = "sensor-01",
+                            Format = DeviceModelMessageSchemaFormat.JSON,
+                            Fields = new Dictionary<string, DeviceModelMessageSchemaType>()
+                            {
+                                {"temp", new DeviceModelMessageSchemaType() }
+                            }
+                        }
+                    }
+                }
+            };   
+
+            var overrideInfo = new DeviceModelOverride();
+
+            // Act
+            var result = this.target.Generate(source, overrideInfo);
+
+            // Assert
+            Assert.NotEmpty(result.Telemetry);
+            for (var i = 0; i < result.Telemetry.Count; i++)
+            {
+                Assert.Equal(result.Telemetry[i].MessageTemplate, source.Telemetry[i].MessageTemplate);
```

Comment:
when using Assert.Equal, the first parameter is the "expected" value, and the second is the "actual".  Invert the 2 params, so that failed assertions will show the correct error message